mne/preprocessing/tests/test_ica.py

```diff
@@ -911,4 +918,64 @@ def test_ica_ctf():
             ica.get_sources(inst)
 
 
+@requires_sklearn
+@testing.requires_testing_data
+def test_ica_eeg():
+    method = 'fastica'
+
+    raw_fif = read_raw_fif(fif_fname, preload=True)
+    raw_eeglab = read_raw_eeglab(input_fname=eeglab_fname,
+                                 montage=eeglab_montage, preload=True)
+
+    for raw in [raw_eeglab, raw_fif]:
+        events = make_fixed_length_events(raw, 99999)
+        picks_meg = pick_types(raw.info, meg=True, eeg=False)
+        picks_eeg = pick_types(raw.info, meg=False, eeg=True)
+        picks_all = pick_types(raw.info, meg=True, eeg=True)
+        epochs = Epochs(raw, events, None, -0.2, 0.2, preload=True)
+        evoked = epochs.average()
+
+        for picks in [picks_meg, picks_eeg, picks_all]:
+            if len(picks) == 0:
+                continue
+            # test fit
+            for inst in [raw, epochs]:
+                ica = ICA(n_components=2, random_state=0, max_iter=2,
+                          method=method)
+                with warnings.catch_warnings(record=True):  # convergence
+                    ica.fit(inst, picks=picks)
+
+            # test apply and get_sources
+            for inst in [raw, epochs, evoked]:
+                ica.apply(inst)
+                ica.get_sources(inst)
+
+    raw = read_raw_ctf(ctf_fname2,  preload=True)
+    events = make_fixed_length_events(raw, 99999)
+    picks_meg = pick_types(raw.info, meg=True, eeg=False)
+    picks_eeg = pick_types(raw.info, meg=False, eeg=True)
+    picks_all = pick_types(raw.info, meg=True, eeg=True)
+    for comp in [0, 1]:
+        raw.apply_gradient_compensation(comp)
+        epochs = Epochs(raw, events, None, -0.2, 0.2, preload=True)
+        evoked = epochs.average()
+
+        for picks in [picks_meg, picks_eeg, picks_all]:
+            if len(picks) == 0:
+                continue
+
+            # test fit
+            for inst in [raw, epochs]:
+                ica = ICA(n_components=2, random_state=0, max_iter=2,
+                          method=method)
+                with warnings.catch_warnings(record=True):  # convergence
+                    ica.fit(inst)
+
+            # test apply and get_sources
+            for inst in [raw, epochs, evoked]:
+                ica.apply(inst)
+                ica.get_sources(inst)
+
+
+
```

Comment:
This displeases Travis:

https://travis-ci.org/mne-tools/mne-python/jobs/389323987#L3171

Reply:
opps